test/cli.js

```diff
@@ -47,4 +47,24 @@ tape('CLI', function (t) {
     spt.succeeds();
     spt.end();
   });
+
+  t.test('standard json', function (st) {
+    var input = {
+      'language': 'Solidity',
+      'sources': {
+        'Contract.sol': {
+          'content': 'pragma solidity ^0.4.0; contract Contract { function f() {} }'
+        }
+      }
+    };
+    var spt = spawn(st, './solcjs --standard-json');
+    spt.stdin.setEncoding('utf-8');
+    spt.stdin.write(JSON.stringify(input), function () {
+      spt.stderr.empty();
+      spt.stdout.match(/Contract.sol/);
+      spt.stdout.match(/userdoc/);
+      spt.succeeds();
+      spt.end();
+    });
```

Comment:
You might need `grep.stdin.end();` around here.

Reply:
This version uses the callback of write which is executed when the stream was pushed. If you look at the commit history, all combinations were tried (pause, end, etc.) and none of them work on Travis. All of them work for me locally.

Any help is appreciated, it is really frustrating at this point.

Reply:
Oh yeah you've already tried at https://github.com/ethereum/solc-js/pull/127/commits/5bef54c21570c412c39e00b7486cbb9fd81ab39d. This behaviour is really wired indeed.

I also tried for 20 times and it turned out `--standard-json` does not work on ci in the first place:
```console
$ solc@0.4.17 test /home/travis/build/watilde/solc-js
> echo {} | ./solcjs --standard-json
Empty input was read
```

Here is the actual log:
https://travis-ci.org/watilde/solc-js/jobs/281608744#L698-L702


I will have a look at `solc-bin`.

Reply:
Probably there is no way around changing solcjs to be asynchronous, because that hack to read stdin may or may not work. Would be nice finding a hack so it can be incrementally developed, i.e. fixing this PR and than changing solcjs entirely to be asynchronous.

Reply:
hey guys, added a fix here https://github.com/ethereum/solc-js/pull/229
seems to do what is needed
any chance you could take a look?
cheers

Reply:
I've pushed it here but it doesn't seem to fix it :(

Reply:
hi, thanks for trying, you most likely need to remove https://github.com/ethereum/solc-js/blob/93292b68f54ec76ae4339de6ae6c68b6cb029c22/solcjs#L58-L62 

here is my build of it `https://travis-ci.org/tzapu/solc-js/builds/394852856`, the fail was because of coveralls, as i don t have it setup/access

Reply:
So apparently `readFileSync(0)` works from node.js 5.x, but we still support (for some reason) 4.x

Reply:
Unfortunately this simple solutions fails on 4.x :(

Reply:
damn it :( i m out of ideas

Reply:
I'm trying to get rid of Node 4.x, unless there is a compelling reason to keep it. Please check #263